Examples/Detectors/Geant4Detector/include/ActsExamples/Geant4Detector/Geant4Detector.hpp

```diff
@@ -32,18 +33,19 @@ class Detector;
 
 namespace ActsExamples {
 class IContextDecorator;
+}  // namespace ActsExamples
 
-namespace Geant4 {
+namespace ActsExamples::Geant4 {
 
-struct Geant4Detector {
-  using DetectorElements =
-      std::vector<std::shared_ptr<Acts::Geant4DetectorElement>>;
+struct Geant4Detector : public DetectorCommons::Detector {
+  using TrackingGeometryPtr = std::shared_ptr<const Acts::TrackingGeometry>;
```

Comment:
Not the only place but I'm personally not a huge fan of these typedefs, as it obscures meaningful details.